lib/commands/actions.js

```diff
@@ -169,33 +172,68 @@ commands.fingerprint = async function (fingerprintId) {
   await this.adb.fingerprint(fingerprintId);
 };
 
+async function storeScreenshotDataWithAdbShell (adb, opts, localFile) {
+  let pngDir = opts.androidScreenshotPath || '/data/local/tmp/';
+  const png = path.posix.resolve(pngDir, 'screenshot.png');
+  let cmd = ['/system/bin/rm', `${png};`, '/system/bin/screencap', '-p', png];
+  await adb.shell(cmd);
+  await adb.pull(png, localFile);
+}
+
+async function storeScreenshotDataWithAdbExecOut (adb, localFile) {
+  let {stdout} = await exec(adb.executable.path, ['exec-out', '/system/bin/screencap -p'],
+                            {encoding: 'binary'});
+  const signature = new Buffer(stdout, 'binary').toString('hex', 0, PNG_MAGIC_LENGTH);
+  if (signature === PNG_MAGIC) {
+    await fs.writeFile(localFile, stdout, 'binary');
+  } else {
+    throw new Error(`Command output is not a valid PNG file (${signature} !== ${PNG_MAGIC})`);
+  }
+}
+
 commands.getScreenshot = async function () {
   let localFile = temp.path({prefix: 'appium', suffix: '.png'});
-  let pngDir = this.opts.androidScreenshotPath || '/data/local/tmp/';
-  const png = path.posix.resolve(pngDir, 'screenshot.png');
-  let cmd =  ['/system/bin/rm', `${png};`, '/system/bin/screencap', '-p', png];
-  await this.adb.shell(cmd);
   if (await fs.exists(localFile)) {
     await fs.unlink(localFile);
   }
-  await this.adb.pull(png, localFile);
-  let image = await jimp.read(localFile);
-  if (await this.adb.getApiLevel() < 23) {
-    // Android bug 8433742 - rotate screenshot if screen is rotated
-    let screenOrientation = await this.adb.getScreenOrientation();
+  if (typeof this._apiLevel === 'undefined') {
```

Comment:
This is already done at the `appium-adb` level (https://github.com/appium/appium-adb/blob/master/lib/tools/adb-commands.js#L30-L40).